metrics.py

```
@@ -14,12 +14,40 @@
 
 import datadog
 import os
+import boto3
 from common import *
+from botocore.exceptions import ClientError
 
+def get_datadog_api_key():
+    secret_name = os.environ.get("DATADOG_API_KEY_SECRET_NAME")
+    region_name = os.environ.get("AWS_REGION", "")
+
+    if not region_name:
+        print("Can't fetch the current AWS region")
+        return
+
+    session = boto3.session.Session()
+    client = session.client(
+        service_name='secretsmanager',
+        region_name=region_name
+    )
+
+    try:
+        get_secret_value_response = client.get_secret_value(
+            SecretId=secret_name
+        )
+    except ClientError as e:
+        print("Can't fetch the secret. Error: {}".format(e.response['Error']['Code']))
+        return
+
+    return get_secret_value_response['SecretString']
 
 def send(env, bucket, key, status):
-    if "DATADOG_API_KEY" in os.environ:
-        datadog.initialize()  # by default uses DATADOG_API_KEY
+    datadog_api_key = get_datadog_api_key()
+    av_datadog_integration_enabled = os.environ.get("AV_DATADOG_INTEGRATION_ENABLED", "false") == "true"
+
+    if datadog_api_key and av_datadog_integration_enabled:
+        datadog.initialize(api_key=datadog_api_key)  # by default uses DATADOG_API_KEY
```

I think you should fetch the key here only if it's enabled. Let the lambda fail if it can't get the token so it can retry verifying the file and try to alert again.